lib/utils.ts

```diff
@@ -117,4 +117,25 @@ function formatActionName(method: string, key?: OnyxKey): string {
     return key ? `${method.toUpperCase()}/${key}` : method.toUpperCase();
 }
 
-export default {isEmptyObject, fastMerge, formatActionName, removeNestedNullValues};
+/** validate that the update and the existing value are compatible */
+function checkCompatibilityWithExistingValue(value: unknown, existingValue: unknown): {isCompatible: boolean; existingValueType?: string; newValueType?: string} {
+    if (!existingValue || !value) {
+        return {
+            isCompatible: true,
+        };
+    }
+    const existingValueType = Array.isArray(existingValue) ? 'array' : 'non-array';
```

Comment:
Should we stick to string, or add a type.

```
type ValueType = 'array' | 'object' | 'other';

const existingValueType: ValueType = Array.isArray(existingValue) ? 'array' : (typeof existingValue === 'object' ? 'object' : 'other');


```

Reply:
I would rather stick to a string, and we can use enum/constants if we need to implement some logic around that. It's more flexible

Reply:
I would also let @roryabraham weigh in on this.